contents/blog/clickhouse-vs-druid.md

```diff
@@ -0,0 +1,187 @@
+
+- **Apache Zookeeper:** ClickHouse does admit a flaw in its single executable promise. If the database is expected to have replicas, it requires Apache Zookeeper to manage the redundancies. However, this is a rather thin layer atop of a ClickHouse cluster, and something that most databases require.
+
+- **Caches:** ClickHouse boasts [different types of caches](https://clickhouse.com/docs/zh/operations/caches/) for discrete optimizations. ClickHouse’s MergeTree table family has unique caches that improve data fetches.
+
+- **Dedicated Engines:** ClickHouse boasts a number of dedicated engines that utilize parallel calculations to expedite data crunching. For instance, the SummingMergeTree Engine is able to sum values throughout a database significantly faster than a linear analog. This makes Clickhouse apt for analytics products that need to do a lot of algebraic math across vast datasets.
+
+### How Druid works
+
+Druid is all about modularity. Druid separates the query, data, and storage nodes into three separate machines. Does that mean there is extra latency? By the Druid’s team’s own concession, an unapologetic yes. Druid’s argument is that intra-database latency is justifiable for the customizable and precision benefits of its approach. 
+
+Let’s take a look at a Druid setup: 
+
+![Druid architecture diagram](../images/blog/clickhouse-vs-druid/druid-architecture.png)  
+
+Druid is built for massive applications with unique data streaming needs. For instance, imagine an application that needs to take the last stock trade into account before returning a value. 
+
+ClickHouse would stumble with this – ClickHouse doesn’t guarantee newly ingested data is included in the next query result. Druid, meanwhile, does – efficiently, too, by storing the newly streamed data temporarily in the data nodes whilst simultaneously packing and shipping it off to deep storage. 
+
+Remember that analogy between the mansion and master-plan community? For Druid, the query nodes are the roads, the data nodes are the houses, and storage is the sprawling, shared lake. 
+
+#### Cattle vs Pets
+
+Druid expects you to treat its nodes like cattle, constantly removing and adding nodes at a whim. You would never do that with ClickHouse instances since each ClickHouse instance likely plays a critical role in your application’s stability. Druid, meanwhile, expects you to utilize up to ten thousand data nodes, each which could be taken offline and maintained without impacting the application.  
+
+Of course, this means that Druid requires more maintenance work. **Much** more maintenance work. Not just that, but server costs, too. Druid is built for large enterprise clients with complex needs and massive financial backing to support the architecture. ClickHouse, meanwhile, is a one-size-fits-most solution; scalable, just not in the same multidimensional, multi-structural way as Druid is.  
```

Comment:
Note you may upset some people with this. If done well treating your infrastructure like cattle can actually be easier to operate as compared to treating your infra as pets. The criticism that you may get is that we are doing things wrong here. The work that infra is working on currently is to make our clusters more like cattle than pets

Reply:
Are there any changes you'd suggest, such as a note to say we recognise the argument to be made here? I can live with a little controversy, but it normally makes sense to acknowledge it!